src/test/java/org/opensrp/repository/postgres/TaskRepositoryTest.java

```diff
@@ -228,4 +228,28 @@ public void testFindAllIdsShouldLimitByGivenParam() {
 		assertEquals(1542027762554l, idsModel.getRight().longValue());
 	}
 
+	public void testGetTasksByOwnerAndPlan() {
+		List<Task> tasks = taskRepository.getTasksByPlanAndOwner("IRS_2018_S1", "demouser", 0);
+		assertEquals(1, tasks.size());
+		assertEquals("tsk11231jh22", tasks.get(0).getIdentifier());
+
+		Task task = new Task();
+		task.setIdentifier("tsk-2332-j");
+		task.setPlanIdentifier("IRS_2018_S");
+		task.setGroupIdentifier("2018_IRS-3734");
+		task.setBusinessStatus("Not Visited");
+		task.setStatus(TaskStatus.DRAFT);
+		task.setServerVersion(System.currentTimeMillis());
+		taskRepository.add(task);
+
+		assertEquals(2, taskRepository.getTasksByPlanAndOwner("IRS_2018_S1", "demouser", 0).size());
```

Comment:
Why does the tasks increase to 2 considering the task created above does not have an `owner` so should not be returned.

Here we expect a list of size 1.

Set the owner then update the task and then assert that list returned has 2

Reply:
Cool